libs/ui-components/src/components/Fleet/CreateFleet/utils.ts

```diff
@@ -72,6 +80,103 @@ export const getValidationSchema = (t: TFunction) => {
   });
 };
 
+export const getFleetPatches = (currentFleet: Fleet, updatedFleet: FleetFormValues) => {
+  let allPatches: PatchRequest = [];
+
+  // Fleet labels
+  const currentLabels = currentFleet.metadata.labels || {};
+  const updatedLabels = updatedFleet.fleetLabels || {};
+
+  const fleetLabelPatches = getLabelPatches('/metadata/labels', currentLabels, updatedLabels);
+  allPatches = allPatches.concat(fleetLabelPatches);
+
+  // Device label selector
+  const currentDeviceSelectLabels = currentFleet.spec.selector?.matchLabels || {};
+  const updatedDeviceSelectLabels = updatedFleet.labels || {};
+
+  const deviceSelectLabelPatches = getLabelPatches(
+    '/spec/selector/matchLabels',
```

Comment:
`spec/selector` may be undefined which would break the patch

Reply:
Good catch, it's now taken care of